### PR TITLE
[FIX] web_editor: duplicate attachments in html field

### PR DESCRIPTION
**Pull request by cammarosano**
Before this commit, when an image was cropped in a record on which multiple users were collaborating, multiple attachments would be created for the same image (one per user).

Multiple attachments for the same image would also be created if the user uploaded the same image multiple times.

This commit makes sure that an existing attachment for the same field values is reused, instead of creating a new one.

task-3524601

### CODE REVIEW

File: addons/web_editor/controllers/main.py [hunk 1]
```diff
@@ -84,6 +84,27 @@ def handle_history_divergence(record, html_field_name, vals):
     # Save only the latest id.
     vals[html_field_name] = incoming_html[0:incoming_history_matches.start(1)] + last_step_id + incoming_html[incoming_history_matches.end(1):]
 
+def get_existing_attachment(ir_attachment, vals):
```
Comment (Goaman) [inline addons/web_editor/controllers/main.py:87]: rename `ir_attachment` to `IrAttachment`

File: addons/web_editor/controllers/main.py [hunk 2]
```diff
@@ -84,6 +84,27 @@ def handle_history_divergence(record, html_field_name, vals):
...
+    """
+    Check if an attachment already exists for the same vals. Return it if
+    so, None otherwise.
+    """
+    fields = dict(vals)
+    fields['res_id'] = fields.get('res_id') or 0
```
Comment (Goaman) [inline addons/web_editor/controllers/main.py:93]: Add a comment to explain why 0.

File: addons/web_editor/controllers/main.py [hunk 3]
```diff
@@ -84,6 +84,27 @@ def handle_history_divergence(record, html_field_name, vals):
...
+    raw, datas = fields.pop('raw', None), fields.pop('datas', None)
+    domain = [(field, '=', value) for field, value in fields.items()]
+    if raw or datas:
+        domain.append(('checksum', '=', ir_attachment._compute_checksum(raw or b64decode(datas))))
```
Comment (Goaman) [inline addons/web_editor/controllers/main.py:97]: Make a guard in the beginning of the method
```py
    if not (raw or datas):
        return None
```

File: addons/web_editor/controllers/main.py [hunk 4]
```diff
@@ -84,6 +84,27 @@ def handle_history_divergence(record, html_field_name, vals):
...
+    try:
+        # Might raise AccessError.
+        existing_attachment = ir_attachment.search(domain, limit=1)
+        if existing_attachment:
+            # Might raise AccessError or UserError.
+            existing_attachment.check('read')
+        return existing_attachment or None
+    except (AccessError, UserError):
+        return None
```
Comment (Goaman) [inline addons/web_editor/controllers/main.py:106]: You should not need to check for access when using the search

File: addons/web_editor/controllers/main.py [hunk 5]
```diff
@@ -352,6 +373,8 @@ def get_image_info(self, src=''):
             'original': (attachment.original_id or attachment).read(['id', 'image_src', 'mimetype'])[0],
         }
 
+
+
```
Comment (Goaman) [inline addons/web_editor/controllers/main.py:377]: 👀

Comment (cammarosano) [inline addons/web_editor/controllers/main.py:377]: Evil empty lines

**Pull request closed without merging**